Test/Integration/Product/ReplicaIndexingTest.php

```diff
@@ -0,0 +1,333 @@
+        $this->replicaManager = $this->objectManager->get(ReplicaManagerInterface::class);
+        $this->indicesConfigurator = $this->objectManager->get(IndicesConfigurator::class);
+        $this->indexSuffix = 'products';
+    }
+
+    protected function getIndexName(string $storeIndexPart): string
+    {
+        return $this->indexPrefix . $storeIndexPart . $this->indexSuffix;
+    }
+
+    public function processFullReindexProducts(): void
+    {
+        $this->processFullReindex($this->productIndexer, $this->indexSuffix);
+    }
+
+    public function testReplicaLimits()
+    {
+        $this->assertEquals(20, $this->replicaManager->getMaxVirtualReplicasPerIndex());
+    }
+
+    /**
+     * @magentoConfigFixture current_store algoliasearch_instant/instant/is_instant_enabled 1
+     */
+    public function testStandardReplicaConfig(): void
+    {
+        $sortAttr = 'created_at';
+        $sortDir = 'desc';
+        $this->assertSortingAttribute($sortAttr, $sortDir);
+
+        $this->indicesConfigurator->saveConfigurationToAlgolia(1);
+        $this->algoliaHelper->waitLastTask();
+
+        // Assert replica config created
+        $indexName = $this->getIndexName('default_');
+        $currentSettings = $this->algoliaHelper->getSettings($indexName);
+        $this->assertArrayHasKey('replicas', $currentSettings);
+
+        $sortIndexName = $indexName . '_' . $sortAttr . '_' . $sortDir;
+
+        $this->assertTrue($this->isStandardReplica($currentSettings['replicas'], $sortIndexName));
+        $this->assertFalse($this->isVirtualReplica($currentSettings['replicas'], $sortIndexName));
+
+        $replicaSettings = $this->assertReplicaIndexExists($indexName, $sortIndexName);
+        $this->assertStandardReplicaRanking($replicaSettings, "$sortDir($sortAttr)");
+    }
+
+    /**
+     * This test involves verifying modifications in the database
+     * so it must be responsible for its own set up and tear down
+     * @magentoDbIsolation disabled
+     * @group virtual
+     */
+    public function testVirtualReplicaConfig(): void
+    {
+        $indexName = $this->getIndexName('default_');
+        $ogSortingState = $this->configHelper->getSorting();
+
+        $productHelper = $this->objectManager->get(ProductHelper::class);
+        $sortAttr = 'color';
+        $sortDir = 'asc';
+        $attributes = $productHelper->getAllAttributes();
+        $this->assertArrayHasKey($sortAttr, $attributes);
+
+        $this->assertNoSortingAttribute($sortAttr, $sortDir);
+
+        $sorting = $this->configHelper->getSorting();
+        $sorting[] = [
+            'attribute'      => $sortAttr,
+            'sort'           => $sortDir,
+            'sortLabel'      => $sortAttr,
+            'virtualReplica' => 1
+        ];
+        $this->configHelper->setSorting($sorting);
+
+        $this->assertConfigInDb(ConfigHelper::SORTING_INDICES, json_encode($sorting));
+
+        $this->refreshConfigFromDb();
```

Comment:
It's not clear to me why this needs to be run here, curious about it

Reply:
This was needed because while testing the writes to the database I discovered that the in-memory config cache for duration of the PHPUnit test function is not updated so calls to ScopeConfigInterface were providing stale data. I can refresh it with `ReinitableConfigInterface` but then it wipes out the earlier bootstrapping which configures via environment variables and also initializes the index name - a reference to which must be preserved before the test can finish.